Copy paste to clipboard button added. Copy paste to clipboard shortcu…

Ribas160 opened the pull request:
This PR closes #1390 and closes #12

## Changes
* Copy to clipboard button added to the top right corner of the paste message block
* Copy to clipboard using keyboard shortcut Ctrl+c/Ctrl+v feature added.
* Added copy to clipboard shortcut hint above the paste message block
* Copy to clipboard shortcut hint and copy to clipboard notifications added to the dictionary files.

Preview
<img width="1440" alt="image" src="https://github.com/user-attachments/assets/c7c162de-f7d9-491a-8b4d-4640d3d7b6ab" />

Test site: https://test.ribas160.ru/privatebin/

elrido commented on i18n/ar.json:
```diff
@@ -219,5 +219,7 @@
-    "Error decompressing paste, your browser does not support WebAssembly. Please use another browser to view this paste.": "Error decompressing paste, your browser does not support WebAssembly. Please use another browser to view this paste."
+    "Error decompressing paste, your browser does not support WebAssembly. Please use another browser to view this paste.": "Error decompressing paste, your browser does not support WebAssembly. Please use another browser to view this paste.",
+    "Paste copied to clipboard": "Paste copied to clipboard",
+    "To copy paste press on the copy button or use the clipboard shortcut Ctrl+c/Cmd+c": "To copy paste press on the copy button or use the clipboard shortcut Ctrl+c/Cmd+c"
```
Over in #1463 I propose to make messages referring to keyboard-shortcuts more accessible by wrapping the keys into kbd-tags. Could we apply the same here, too? Example:
https://github.com/PrivateBin/PrivateBin/blob/5aa7a95841779633e6cb80aea343991144dfc4e9/i18n/ar.json#L154

elrido reviewed (changes requested):
Thank you for tackling this, a very handy feature!

If you could re-use those icons included with the bootstrap CSS themes, as suggested in the inline-comments, you may be able to reduce the custom CSS changes necessary.

Purely as a wishlist item: While you are adding such buttons and logic to copy the whole paste text, could you also add one to the created paste link, as suggested in #12?

Please do add an ADDED-entry to the CHANGELOG.md in the future release and consider adding yourself to the CREDITS.md.

Ribas160 commented on js/privatebin.js:
```diff
@@ -757,34 +757,34 @@ jQuery.PrivateBin = (function($, RawDeflate) {
             }
 
-            // messageID may contain links, but should be from a trusted source (code or translation JSON files)
-            let containsLinks = args[0].indexOf('<a') !== -1;
```
The previous code allowed some html tags in the translation text only if it contained a link. So *kbd* tags were printed as text, not as an html tag. I added the code that allows all html tags that passed DOMPurify filters.

Ribas160 commented:
@elrido 
I made the changes you suggested.

Regarding CREDITS.md I would be pleased to be on this list. At the moment my contribution to the project is not very big, should I add myself to the *Active contributors* list or add to *Past contributions* for now?

As I understood in #12, we want to add a button to copy the link to the clipboard and save the current ctrl+c shortcut.
Now I'm thinking about what this button could looks like.
It could be a button in the top navigation menu. It could be a text button next to the "Delete data" link, but I'm afraid that on mobile phones it's easy to miss the click and delete the paste instead of copying. Or it could be like in the screenshot. What is your opinion?

<img width="1440" alt="image" src="https://github.com/user-attachments/assets/de288b06-e101-40ef-a456-44615c53a59b" />

elrido commented:
> @elrido I made the changes you suggested.

Thank you, it looks good! I'll approve the current set of changes, but let the PR unmerged for now, in case any one else wants to comment or if you'd want to work on #12 in this branch.

> Regarding CREDITS.md I would be pleased to be on this list. At the moment my contribution to the project is not very big, should I add myself to the _Active contributors_ list or add to _Past contributions_ for now?

I'd add it to the past ones, which are mostly the occasional features that got added. The active ones are intended to be the ones that are frequently participating on the project on an ongoing basis. Of course, if you keep participating, we are more than happy to "upgrade" you to a maintainer - we are looking for more active collaborators.

> As I understood in #12, we want to add a button to copy the link to the clipboard and save the current ctrl+c shortcut. Now I'm thinking about what this button could looks like. It could be a button in the top navigation menu. It could be a text button next to the "Delete data" link, but I'm afraid that on mobile phones it's easy to miss the click and delete the paste instead of copying. Or it could be like in the screenshot. What is your opinion?

Yes, I would make it a separate button, above the message box as you suggest. And on our side, we should also consider turning that delete link into a button, too. As a pure link it is easy to miss and also not mobile friendly. Something like that was already suggested in #268. But that is for a separate pull request later on.

Ribas160 commented:
@elrido 
I amended the previous commit, added css to support RTL.

In the new commit I added the copy link button as on the screenshot I sent you before.

You can see a demo on the my test site: https://test.ribas160.ru/privatebin/

Ribas160 commented on css/bootstrap/privatebin.css:
```diff
@@ -143,34 +143,23 @@ html[dir="rtl"] #deletelink, html[dir="rtl"] #qrcodemodalClose {
-
-#copyIcon:hover {
-	opacity: 1;
+html[dir="rtl"] #prettyMessageCopyBtn {
```
I amended the previous commit, added css for RTL support. Similar change I did for the bootstrap5 template.

elrido reviewed (approved):
Thank you, that is excellent work!

rugk commented:
> Tastenkombination Strg+c/Cmd+c

here:
![grafik](https://github.com/user-attachments/assets/d3168e43-d0bc-4240-88ba-e4cf21b339af)

You may detect the OS to show the proper command (actually only MacOS AFAIK has <kbd>Cmd</kbd>), the other one may just confuse/is not relevant to show. 
This would be a useful addition, IMHO. 
Technically may be harder [but this seems okay](https://stackoverflow.com/a/73619128/5008962) [through browser support says it is still experimental](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorUAData/platform).

Ribas160 commented:
@rugk 
That's an interesting idea, but I'm thinking how to do it better.
We could replace **Ctrl+c/Cmd+c** with **%s+c** and just insert the relevant button key into the text. But if someone decides to render this translation directly in the template, that means they'll have to create a PHP logic to detect the system.
Maybe we can use just **Ctrl+c** like we do for copy link shortcut.

MickGe commented:
Hi,

I don't understand why the copy button is hidden if the paste is made with markdown.

https://github.com/PrivateBin/PrivateBin/blob/master/js/privatebin.js#L2767

rugk commented:
Hmm, it should not hide the buttom, indeed does not sound logical. Could you create a new issue please?

rugk commented:
Also maybe for my cmd/ctrl proposal before, we ahould create a new issue.

Ribas160 commented:
This happens because the copy button is inside the `#prettymessage` block where the app renders the saved paste text
https://github.com/PrivateBin/PrivateBin/blob/0a37a884e24af7b8391e7861e7767e0572d021ff/tpl/bootstrap.php#L591-L597

But when markdown format enabled the app renders the saved paste inside the `#plaintext` block and hides the `#prettymessage`.
https://github.com/PrivateBin/PrivateBin/blob/0a37a884e24af7b8391e7861e7767e0572d021ff/tpl/bootstrap.php#L598

I suggest to create some common container for `#prettymessage` and `#plaintext` and put the copy button next to these blocks.

But anyway it is a discussion for separated issue. Also during the work with `#plaintext` we may consider suggestions from #1391

rugk commented:
Opened https://github.com/PrivateBin/PrivateBin/issues/1506 for my proposal BTW.

@MickGe you may best open an issue for your idea by yourself.
